slave.mk

```diff
@@ -1208,6 +1210,62 @@ $(DOCKER_LOAD_TARGETS) : $(TARGET_PATH)/%.gz-load : .platform docker-start $$(TA
 ## Installers
 ###############################################################################
 
+
+define rfs_get_installer_dependencies
+$(1)__$($(1)_MACHINE)__rfs.squashfs $(if $($(1)_DEPENDENT_MACHINE),$(1)__$($(1)_DEPENDENT_MACHINE)__rfs.squashfs)
+endef
+
+$(foreach installer, $(SONIC_INSTALLERS), $(eval $(installer)_RFS_DEPENDS=$(call rfs_get_installer_dependencies,$(installer))))
+SONIC_RFS_TARGETS= $(foreach installer, $(SONIC_INSTALLERS), $(call rfs_get_installer_dependencies,$(installer)))
+
+ifeq ($(ENABLE_RFS_SPLIT_BUILD),y)
+SONIC_TARGET_LIST += $(addprefix $(TARGET_PATH)/, $(SONIC_RFS_TARGETS))
+endif
+
+$(addprefix $(TARGET_PATH)/, $(SONIC_RFS_TARGETS)) : $(TARGET_PATH)/% : \
```

Comment:
>SONIC_RFS_TARGETS [](http://example.com/codeflow?start=31&length=17)

Could you make sure the new target `SONIC_RFS_TARGETS` is reused if we build multiple vendors image in series?

We also have reproducible mechanism in rules/*.dep files. Is it possible to define this target's dep file, so the target is reused as cached for other later build jobs?

@xumia to review.

Reply:
The target is reused. I've also added a cache: https://github.com/sonic-net/sonic-buildimage/pull/15924/commits/a880d53d14d593f6739e0ca33d22354157503aec
Since the targets are defined dynamically, I could not add the '.dep' file. Instead, the relevant variables are also set dynamically.